Code/GraphMol/Abbreviations/Abbreviations.cpp

```diff
@@ -76,11 +76,13 @@ void applyMatches(RWMol& mol, const std::vector<AbbreviationMatch>& matches) {
       mol.addBond(oaidx, connectIdx, Bond::BondType::SINGLE);
     }
   }
-  for (unsigned int i = toRemove.size(); i > 0; --i) {
-    if (toRemove[i - 1]) {
-      mol.removeAtom(i - 1);
+  mol.beginBatchEdit();
```

Comment:
As you know I really like RAII, perhaps we could use a C++ context manager as well

```
BatchEdit editor(mol);
for(auto &atm : mol.atoms()) {
    mol.removeAtom(atm)
}
```

```
BatchEdit editor(mol);
try {
for(auto &atm : mol.atoms()) {
      mol.removeAtom(atm)
} catch (...) {
  editor.cancel();
}
}
```

I think this is a lot less error prone than missing a mol.commitBatchEdit();